### PR TITLE
feature/card

### PR DESCRIPTION
**Pull request by castromaciel**
- Add basic carde modes
- Add tests
- Export files

### CODE REVIEW

**Comment by Nidev-lab** [comment]
Add a screenshoot for each pr, please.

**Comment by castromaciel** [comment]
# Card 

### mode:  classic (default)

```tsx
<Card
  title="Classic"
  buttonRight="View more"
  onClickRight={() => console.log('Clicked')}
  description="Lorem ipsum dolor sit amet, adipiscing elit."
  image="https://picsum.photos/id/237/300/300"
  altImage="Alternative text"
/>
```

image

### mode: product

```tsx
<Card
  mode="product"
  title="Product"
  price="1,034.50"
  buttonRight="Buy"
  onClickRight={() => console.log('Clicked')}
  description="Lorem ipsum dolor sit amet, consectetur adipiscing elit."
  image="https://picsum.photos/id/27/2000/2000"
  altImage="Alternative text"
/>
```

image

**Review by Nidev-lab — CHANGES_REQUESTED** [review]

File: src/typinigs.d.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+declare module '*.module.scss' {
+  const content: { [className: string]: string };
+  export default content;
+}
+
+declare module '*.jpg' {
+  const value: any;
+  export default value;
+}
```
[the remark this answers is not in the document]
Comment (castromaciel) [reply src/typinigs.d.ts]: Solved at commit: [fix in the end line at typings file](https://github.com/castromaciel/ui-rollingcode/pull/8/commits/2a0c68d0e739acfb2421ab75b4b00e808455d4e2)

File: src/structure/Card/Card.test.tsx [hunk 2]
```diff
@@ -0,0 +1,48 @@
+/* eslint-disable no-undef */
```
[the remark this answers is not in the document]
Comment (castromaciel) [reply src/structure/Card/Card.test.tsx]: That was i've got the following error

```
'describe' is not defined. eslint (no-undef)
```

I've solved adding

``` json
"env": {
    "jest": true
}
```

 at `.eslintrc.json`

Associate commit: [fix eslint-disable no-undef for tests with jests](https://github.com/castromaciel/ui-rollingcode/pull/8/commits/2f8e1116809d707bdca107b962b4937ab7c6ade4)

File: src/structure/Card/card.module.scss [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+    padding: 0rem 1rem 1rem;
+    display: flex;
+    flex-direction: column;
+    gap: 1rem;
+
+    h2 {
+      margin: 0;
+    }
+  
+
```
[the remark this answers is not in the document]
Comment (castromaciel) [reply src/structure/Card/card.module.scss]: Solved at commit '[fix end line at styles](https://github.com/castromaciel/ui-rollingcode/pull/8/commits/74f9576b4646807204b3bb644306e035048f7bb1)'

File: src/structure/Card/card.module.scss [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+    .description {
+      margin: 0;
+    }
+
+    hr {
+      color: #c82525ce;
+      margin: 0;
+      width: 100%;
+      border: 0.01em solid #c82525ce;
+    }
```
[the remark this answers is not in the document]
Comment (castromaciel) [reply src/structure/Card/card.module.scss:45]: Solved at commit '[fix end line at styles](https://github.com/castromaciel/ui-rollingcode/pull/8/commits/74f9576b4646807204b3bb644306e035048f7bb1)'

File: src/structure/Card/Card.tsx [hunk 5]
```diff
@@ -1,5 +1,79 @@
...
+
+      <p className={`${cardStyles.description}`}>{description}</p>
+
+      {
+        mode === 'classic'
+          ? (
+            <div className={`${cardStyles.buttons_container}`}>
+              { buttonLeft
+                ? <button type="button" className={`${cardStyles.button_left}`} onClick={onClickLeft}>{buttonLeft}</button>
+                : null }
```
[the remark this answers is not in the document]
Comment (castromaciel) [reply src/structure/Card/Card.tsx]: Change at commit: [Add default props](https://github.com/castromaciel/ui-rollingcode/pull/8/commits/0a217b9400799f8781c778fb3cd7643900356421) and [fix validate at button left and colors](https://github.com/castromaciel/ui-rollingcode/pull/8/commits/19cbe8bb872f62839e7c10804012dcccabf75688)

It was my fault, i've confused by [this](https://kentcdodds.com/blog/use-ternaries-rather-than-and-and-in-jsx). But in our case, we don't compare anything with `0` or `''`.

File: src/structure/Card/Card.tsx [hunk 6]
```diff
@@ -1,5 +1,79 @@
...
+  className = '',
+  description,
+  id,
+  image,
+  mode = 'classic',
+  onClickLeft,
+  onClickRight,
+  style,
+  symbol = '$',
+  price = 100,
```
[the remark this answers is not in the document]
Comment (castromaciel) [reply src/structure/Card/Card.tsx]: Solved at commit [Add default props](https://github.com/castromaciel/ui-rollingcode/pull/8/commits/0a217b9400799f8781c778fb3cd7643900356421)

File: src/structure/Card/card.module.scss [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+        max-width: 120px;
+      }
+
+      button {
+        all: unset;
+        font-size: 14px;
+        cursor: pointer;
+        font-weight: bold;
+        padding: 0.5rem 1rem;
+        border: 1px solid #C82526;
```
[the remark this answers is not in the document]
Comment (castromaciel) [reply src/structure/Card/card.module.scss:66]: Okey, i'll be working at that ticket, and when it's finished, I'll make this changes.

Applies at commit: [fix validate at button left and colors](https://github.com/castromaciel/ui-rollingcode/pull/8/commits/19cbe8bb872f62839e7c10804012dcccabf75688)

File: src/structure/Card/Card.tsx [hunk 8]
```diff
@@ -1,5 +1,79 @@
...
+
+export interface CardProps {
+  buttonRight: string
+  description: string
+  image: string
+  title: string
+  onClickRight: MouseEventHandler<HTMLButtonElement>
+  
+  altImage?: string
+  buttonLeft?: string
```
[the remark this answers is not in the document]
Comment (castromaciel) [reply src/structure/Card/Card.tsx]: Solved at [Add default props](https://github.com/castromaciel/ui-rollingcode/pull/8/commits/0a217b9400799f8781c778fb3cd7643900356421).

More than `action` property, I still `onClick`. That's beacause, the action button we define is MouseEventHandler `onClick`

**Review by Nidev-lab — APPROVED** [review]

**Pull request merged**